core/src/com/fallenflame/game/AIController.java

```diff
@@ -1,5 +1,411 @@
+                }
+
+                else if(investigateReached()){
+                    investigationPosition = null;
+                    enemy.setActivated(false);
+                    state = FSMState.WANDER;
+                }
+
+                break;
+
+            default:
+                assert (false);
+                state = FSMState.WANDER;
+                break;
+        }
+    }
+
+    /**
+     * Mark all desirable tiles to move to.
+     *
+     * This method implements pathfinding through the use of goal tiles.
+     * It searches for all desirable tiles to move to (there may be more than
+     * one), and marks each one as a goal. Then, the pathfinding method
+     * getMoveAlongPathToGoalTile() moves the ship towards the closest one.
+     *
+     * POSTCONDITION: There is guaranteed to be at least one goal tile
+     * when completed.
+     */
+    private void markGoalTiles() {
+        boolean setGoal = false;
+        float playerX = player.getX(), playerY = player.getY();
+        int sx, sy;
+
+        switch (state) {
+            case SPAWN:
+                break;
+
+            case WANDER:
+                float x, y;
+                if (enemy.getGoal() == null || random.nextInt(100) < 10) {
+                    x = random.nextFloat() * level.getHeight();
+                    y = random.nextFloat() * level.getWidth();
+                    sx = level.screenToTile(x);
+                    sy = level.screenToTile(y);
+                    if (level.getSafe(sx, sy)) {
+                        enemy.setGoal(x, y);
+                        setGoal = true;
+                    }
+
+                }
+
+                x = enemy.getGoalX();
+                y = enemy.getGoalY();
+                sx = level.screenToTile(x);
+                sy = level.screenToTile(y);
+                if (level.getSafe(sx, sy)) {
+                    setGoal = true;
+                }
+
+                break;
+
+            case CHASE:
+                sx = level.screenToTile(playerX);
+                sy = level.screenToTile(playerY);
+                if(level.getSafe(sx, sy)){
+                    enemy.setGoal(playerX, playerY);
+                    setGoal = true;
+                }
+                break;
+
+            case ATTACK:
+                sx = level.screenToTile(playerX);
+                sy = level.screenToTile(playerY);
+                if(level.getSafe(sx, sy)){
+                    enemy.setGoal(playerX, playerY);
+                    setGoal = true;
+                }
+                break;
+
+            case INVESTIAGE: //investigationPosition must not be null
+                float investX = investigationPosition.x;
+                float investY = investigationPosition.y;
+                sx = level.screenToTile(playerX);
+                sy = level.screenToTile(playerY);
+                if(level.getSafe(sx, sy)){
+                    enemy.setGoal(investX, investY);
+                }
```

Comment:
Probably won't be safe bc flare sticks to wall. Again I don't think we should check safety of goal locations, that should just be for movement